lib/cocina_checker.rb

```diff
@@ -73,4 +90,15 @@ def group_by_unique_values(hash)
       .group_by { |_key, value| value }
       .transform_values { |value| value.map(&:first) }
   end
+
+  # git checkout repo to the given tag, or switch to default branch if none
+  def switch_repo_to_tag(lockfile_path, target)
+    Dir.chdir(lockfile_path.delete_suffix('/Gemfile.lock'))
```

Comment:
i think it might be slightly preferable to do the work you want to do in a block passed to `chdir` -- https://ruby-doc.org/core-3.1.2/Dir.html#method-c-chdir
> If a block is given, it is passed the name of the new current directory, and the block is executed with that as the current directory. The original working directory is restored when the block exits.

that seems to be the pattern used elsewhere:  https://github.com/sul-dlss/sdr-deploy/search?q=chdir